units/provisioning_setup_host.go

```diff
@@ -890,21 +890,21 @@ func writeIcecreamConfig(ctx context.Context, env evergreen.Environment, h *host
 	return nil
 }
 
-func getMostRecentlyAddedDevice(lsblkOutput []string) (string, error) {
+func getMostRecentlyAddedDevice(lsblkOutput []string) (string, string, error) {
 	// lsblk contains at least the header and one device
 	if len(lsblkOutput) < 2 {
-		return "", errors.Errorf("lsblk output is length '%d'", len(lsblkOutput))
+		return "", "", errors.Errorf("lsblk output is length '%d'", len(lsblkOutput))
 	}
 
 	device := lsblkOutput[len(lsblkOutput)-1]
-	deviceNameRegexp, err := regexp.Compile(`^(?P<deviceName>sd[a-z]\d{0,2}|xvd[a-z]|hd[a-z]\d{0,2}|nvme\d{1,2}n1)`)
+	deviceNameRegexp, err := regexp.Compile(`^(?P<deviceName>sd[a-z]\d{0,2}|xvd[a-z]|hd[a-z]\d{0,2}|nvme\d{1,2}n1).*\b(?P<deviceUUID>[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})$`)
```

Comment:
we have a UUID library that should be able to validate this without a regex. https://godoc.org/github.com/google/uuid#UUID

Reply:
I ended up removing the regex

Reply:
I could still validate it, if you see value in it.